src/main/java/net/rptools/maptool/client/functions/LookupTableFunction.java

```diff
@@ -84,367 +90,521 @@ public Object childEvaluate(
-      LookupTable lookupTable = getMaptoolTable(name, function);
-      MD5Key img = lookupTable.getTableImage();
-      // Returning null causes an NPE when output is dumped to chat.
-      return Objects.requireNonNullElse(img, "");
-
+      return getTableImage(function, params);
     } else if ("setTableImage".equalsIgnoreCase(function)) {
+      return setTableImage(function, params);
+    } else if ("copyTable".equalsIgnoreCase(function)) {
+      return copyTable(function, params);
+    } else if ("setTableEntry".equalsIgnoreCase(function)) {
+      return setTableEntry(function, params);
+    } else if ("getTableEntry".equalsIgnoreCase(function)) {
+      return getTableEntry(function, params);
+    } else if ("resetTablePicks".equalsIgnoreCase(function)) {
+      return resetTablePicks(function, params);
+    } else if ("setTablePickOnce".equalsIgnoreCase(function)) {
+      return setTablePickOnce(function, params);
+    } else if ("getTablePickOnce".equalsIgnoreCase(function)) {
+      return getTablePickOnce(function, params);
+    } else if ("getTablePicksLeft".equalsIgnoreCase(function)) {
+      return getTablePicksLeft(function, params);
+    } else { // if tbl, table, tblImage or tableImage
+      return tbl_and_table(function, params);
+    }
+  }
 
-      checkTrusted(function);
-      FunctionUtil.checkNumberParam("setTableImage", params, 2, 2);
-      String name = params.get(0).toString();
-      MD5Key asset = FunctionUtil.getAssetKeyFromString(params.get(1).toString());
-      LookupTable lookupTable = getMaptoolTable(name, function);
-      lookupTable.setTableImage(asset);
-      MapTool.serverCommand().updateCampaign(MapTool.getCampaign().getCampaignProperties());
-      return "";
+  private @Nullable Serializable tbl_and_table(String function, List<Object> params)
+      throws ParserException {
+    FunctionUtil.checkNumberParam(function, params, 1, 3);
+    String name = params.get(0).toString();
 
-    } else if ("copyTable".equalsIgnoreCase(function)) {
+    String roll = null;
+    if (params.size() > 1) {
+      roll = params.get(1).toString().isEmpty() ? null : params.get(1).toString();
+    }
 
-      checkTrusted(function);
-      FunctionUtil.checkNumberParam("copyTable", params, 2, 2);
-      String oldName = params.get(0).toString();
-      String newName = params.get(1).toString();
-      LookupTable oldTable = getMaptoolTable(oldName, function);
-      LookupTable newTable = new LookupTable(oldTable);
-      newTable.setName(newName);
-      MapTool.getCampaign().getLookupTableMap().put(newName, newTable);
-      MapTool.serverCommand().updateCampaign(MapTool.getCampaign().getCampaignProperties());
-      return "";
+    LookupTable lookupTable = checkTableAccess(function, name);
+    LookupEntry result = lookupTable.getLookup(roll);
+    if (result == null) {
+      return null;
+    }
 
-    } else if ("setTableEntry".equalsIgnoreCase(function)) {
+    assert result.getValue() != null;
```

Comment:
If this should never happen, please place a comment describing why so people reading the code know its not just a mistake that should be replaced, as otherwise, it should be a more meaningful error reported.